diff --git a/home/user/apps/deck.gl/modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer-fragment.glsl.js b/./mesh-layer/simple-mesh-layer-fragment.glsl.js
index 4de17146..b1d6b7d0 100644
--- a/home/user/apps/deck.gl/modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer-fragment.glsl.js
+++ b/./mesh-layer/simple-mesh-layer-fragment.glsl.js
@@ -17,6 +17,15 @@ in vec4 vColor;
 out vec4 fragColor;
 
 void main(void) {
+  
+#ifdef MODULE_PBR
+
+  fragColor = vColor * pbr_filterColor(vec4(0));
+  geometry.uv = pbr_vUV;
+  fragColor.a *= opacity;
+
+#else
+
   geometry.uv = vTexCoord;
 
   vec3 normal;
@@ -35,10 +44,12 @@ void main(void) {
     normal = normals_commonspace;
   }
 
-  vec4 color = hasTexture ? texture(sampler, vTexCoord) : vColor;
+  vec4 color = texture(sampler, vTexCoord) * vColor;
   vec3 lightColor = lighting_getLightColor(color.rgb, cameraPosition, position_commonspace.xyz, normal);
   fragColor = vec4(lightColor, color.a * opacity);
 
+#endif
+
   DECKGL_FILTER_COLOR(fragColor, geometry);
 }
 `;
